### PR TITLE
Add description for Azure url format

### PR DESCRIPTION
**Pull request by steffansluis**
- Add a table entry for the URL format to make use of `dvc import-url` with Microsoft Azure Blob Storage.
- Add example for usage of Microsoft Azure Blob Storage with external dependencies.

> > You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.
> 
> ❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.
> 
> 🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.
> 
> Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.
> 
> Thank you for the contribution - we'll try to review it as soon as possible. 🙏
>

### CODE REVIEW

**Comment by efiop** [comment]
Thank you for the docs @steffansluis ! :pray: Could you also add azure to https://dvc.org/doc/user-guide/external-dependencies , please?

**Comment by shcheklein** [comment]
Thanks @steffansluis 🙏

File: content/docs/command-reference/import-url.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ DVC supports several types of (local or) remote locations (protocols):
 | -------- | --------------------------------------------------- | ------------------------------------------ |
 | `local`  | Local path                                          | `/path/to/local/data`                      |
 | `s3`     | Amazon S3                                           | `s3://mybucket/data`                       |
+| `azure`  | Microsoft Azure Blob Storage                        | `azure://my-container-name/path/to/data`   |
 | `gs`     | Google Storage                                      | `gs://mybucket/data`                       |
```
Comment (jorgeorpinel) [inline content/docs/command-reference/import-url.md:59]: Unrelated, I know:
```suggestion
| `gs`     | Google Cloud Storage                                | `gs://mybucket/data`                       |
```

Comment (jorgeorpinel) [reply content/docs/command-reference/import-url.md:59]: My plan was to just commit this but the PR doesn't allow maintainers to push to this branch. would be great if you can apply this @steffansluis

File: content/docs/user-guide/external-dependencies.md [hunk 2]
```diff
@@ -64,6 +65,18 @@ $ dvc run -d s3://mybucket/data.txt \
...
+### Microsoft Azure Blob Storage
+
+```dvc
+$ dvc run -d azure://my-container-name/data.txt \
+          -o data.txt \
+          az storage copy \
+                            -d data.json \
+                            --source-account-name my-account \
+                            --source-container my-container-name \
+                            --source-blob data.txt
```
Comment (jorgeorpinel) [inline content/docs/user-guide/external-dependencies.md:77]: The indentation here seems excessive? Code block lines shouldn't go over 72 characters long, please. How about just align the flags under `copy`? Or go further back if still they go over 72 🙂

Comment (steffansluis) [reply content/docs/user-guide/external-dependencies.md:77]: The indentation is based on the indentation used in the HDFS example. I completely agree that it is excessive, but I figured there was probably a reason for it. I'll move it back a bit so they line up better with `copy`, I did check though and the longest line was 67 chars so it should be fine regardless

**Comment by shcheklein** [comment]
Thanks @steffansluis 🙏

File: content/docs/user-guide/external-dependencies.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md]: Thanks for bringing this to out attention @steffansluis I'll review this in a separate PR 🙂

**Pull request merged**